app/models/user.rb

```diff
@@ -0,0 +1,9 @@
+class User < ApplicationRecord
+  before_save{self.email = email.downcase}
```

Comment:
```suggestion
  before_save{email.downcase!}
```
so sánh giúp a .downcase vs downcase! nhé

Reply:
dowcase trả về một bản sao của str với tất cả các chữ cái viết hoa được thay thế bằng các bản sao chữ thường của chúng. 
dowcase! thì thay thế luôn các giá trị viết hoa bằng viết thường của chuỗi.